src/understand-tidb/the-lifecycle-of-a-statement.md

```diff
@@ -1 +1,67 @@
 # The Lifecycle of a Statement
+
+## MySQL protocol package with command and statement string
+
+After [connecting and getting authenticated](understand-tidb/mysql-protocol-and-session-management.md) the server is in a [statement execution loop](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/server/conn.go#L938) until the client is disconnected.
+
+The [dispatch](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/server/conn.go#L1111) function checks what kind of command was sent through the MySQL protocol and dispatches the matching function, like this snippet:
+```golang
+	switch cmd {
+...
+	case mysql.ComQuit:
+		return io.EOF
+	case mysql.ComInitDB:
+		if err := cc.useDB(ctx, dataStr); err != nil {
+			return err
+		}
+		return cc.writeOK(ctx)
+	case mysql.ComQuery: // Most frequently used command.
+		return cc.handleQuery(ctx, dataStr)
+```
+Where [mysql.ComQuery](https://github.com/pingcap/parser/blob/d4a88481405f8c59d45fc0a9c38ee24d55b9bf49/mysql/const.go#L102) is routed to [handleQuery](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/server/conn.go#L1633), which handles all different non-prepared statements (some commands like change database/schema or ping are handled directly in the dispatch function).
+
+## Statement string to Abstract Syntax Tree
+
+In [handleQuery](https://github.com/pingcap/tidb/blob/30cf15a59db11c34ffe05fc926152a43327eaa61/server/conn.go#L1633) we first needs to [parse](https://github.com/pingcap/parser/blob/d4a88481405f8c59d45fc0a9c38ee24d55b9bf49/yy_parser.go#L137) the statement string, which is done by [the parser, which is in its own repository](https://github.com/pingcap/parser) which is a MySQL compatible parser which parses statements and returns an Abstract Syntax Tree (AST) representing the statement.
```

Comment:
Maybe add a link to `understand-tidb/parser.md` here?

Reply:
Thanks for the suggestion, I added a link to the planner.md